src/main/java/com/abc/hublearning/web/controller/StudentController.java

```diff
@@ -0,0 +1,50 @@
+package com.abc.hublearning.web.controller;
+
+import com.abc.hublearning.web.dto.StudentDto;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RequestMapping("/students")
+@RestController
+public class StudentController {
+
+    @GetMapping
+    public ResponseEntity<Object> getStudents(){
+        return null;
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Object> getById(@PathVariable Long id){
+
+        //Student student = studentService.getById();
+        //return ResponseEntity.ok(student);
+        return null;
+    }
+
+    @PostMapping
+    public ResponseEntity<Object> createStudent(@RequestBody StudentDto studentDto){
+        return null;
+    }
+
+    @PutMapping("{/id}")
+    public ResponseEntity<Object> updateStudent(@RequestBody StudentDto studentDto, @PathVariable long id){
+        //Optional<Student> studentOptional = studentService.findById(id);
+//        if (!studentOptional.isPresent()) {
+//            return ResponseEntity.notFound().build();
```

Comment:
In ideea rezolvari cazurilor exceptionale vreau sa imi pui la response entity un body. Faci o clasa RestException cu campurile:
long code, string message

Reply:
Am creat clasa "RestException" si am incercat sa fac si o clasa custom de exceptie(ResourceNotFoundException) pentru a vedea daca merge dar am doua probleme desi intoarce exceptia in postman:
1. la metoda care handluieste 'ResourceNotFoundException' din ControllerAdvice cand returnez un ResponseEntity  trebuie sa hardcodez parametrul 2 ca fiind HttpStatus.NOT_FOUND. Nu pare a fi foarte elegant. Probabil nu am inteles eu flow-ul corect.

2. desi imi intoarce exceptia corect, pe langa ea imi da un intreg stacktrace in postman:
    "cause": null,
    "stackTrace": [
        {
            "methodName": "handleEntityNotFound",
            "fileName": "RestResponseEntityExceptionHandler.java",
            "lineNumber": 27,
            "className": "com.abc.hublearning.web.error.RestResponseEntityExceptionHandler",
            "nativeMethod": false
        },
        {
            "methodName": "invoke0",
            "fileName": "NativeMethodAccessorImpl.java",
            "lineNumber": -2,
            "className": "sun.reflect.NativeMethodAccessorImpl",
            "nativeMethod": true
        },............
Dupa MESAJUL ASTA LUNG imi apare si mesajul din clasa RestException:
"code": 404,
    "message": "resource doesn't exist",
    "localizedMessage": "resource doesn't exist",
    "suppressed": []                  E ok?

Scuze pentru mesajul asta foarte lung!!!